src/hardware/mouse/mouseif_dos_driver.cpp

```diff
@@ -853,53 +860,67 @@ static void limit_coordinates()
 	limit(pos_y, state.minpos_y, state.maxpos_y);
 }
 
-static void update_mickeys_on_move(const float x_rel, const float y_rel)
+static void update_mickeys_on_move(float& dx, float& dy,
+                                   const float x_rel,
+                                   const float y_rel)
 {
-	auto update = [](int16_t &counter, float &delta, const float rel) {
-		delta += rel;
-
-		// Check if movement is significant enough
-		const auto d = static_cast<int16_t>(std::lround(delta));
-		if (d == 0)
-			return;
-
-		// Consume part of delta to increase/decrease the counter
-		delta -= d;
-		int32_t counter_big = counter + d;
-
-		// Handle counter wrap around int16_t limits
-		if (counter_big > INT16_MAX)
-			counter_big -= UINT16_MAX + 1;
-		else if (counter_big < INT16_MIN)
-			counter_big += UINT16_MAX + 1;
-
-		counter = static_cast<int16_t>(counter_big);
+	auto calculate_d = [](const float rel,
+	                      const float pixel_per_mickey,
+	                      const float sensitivity_coeff) {
+		float d = rel * pixel_per_mickey;
+		if (is_input_raw) {
+			// For RAW input, apply a DOSBox mouse acceleration
+			if ((fabs(rel) > 1.0f) || (sensitivity_coeff < 1.0f)) {
+				d *= sensitivity_coeff;
+			}
+			// TODO: add an alternative calculation (configurable),
+			// reuse MOUSE_GetBallisticsCoeff for DOS driver
+		} else {
+			// If host OS mouse acceleration is used, do not apply
+			// our own acceleration; just multiply by the sensitivity
+			d *= sensitivity_coeff;
+		}
```

Comment:
Maybe just one branch?
```c++
if (!is_input_raw || (fabs(rel) > 1.0f) || (sensitivity_coeff < 1.0f)) {
				d *= sensitivity_coeff;
}
return d;
```

Reply:
OK, changed. This part will have to be changed nevertheless once alternative acceleration code is provided.